tests/lightning_qubit/test_adjoint_jacobian_class.py

```diff
@@ -138,6 +156,35 @@ def test_empty_measurements(self, lightning_sv):
         jac = self.calculate_jacobian(lightning_sv(num_wires=3), tape)
         assert len(jac) == 0
 
+    def test_empty_trainable_params(self, lightning_sv):
+        """Tests if an empty array is returned when the number trainable params is zero."""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.X(wires=[0])
+            qml.expval(qml.PauliZ(0))
+
+        jac = self.calculate_jacobian(lightning_sv(num_wires=3), tape)
+        assert len(jac) == 0
+
+    def test_not_expectation_return_type(self, lightning_sv):
+        """Tests if an empty array is returned when the number trainable params is zero."""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.X(wires=[0])
+            qml.RX(0.4, wires=[0])
+            qml.var(qml.PauliZ(1))
```

Comment:
I will check if it is possible to change all the `QuatumTape` by `QuantumScript` because all this file follows the same strategy. Do you recommend changing it in this PR, or will it be better to open an independent PR and change `QuantumScript` there? @mlxd  :thinking:

Reply:
I think a separate PR is fine, but I wouldn't backlog this item for future work, and aim to get it done as a direct follow on to this PR.